### ISSUE TITLE
Cannot re-produce the results

### ISSUE THREAD

**Issue by yxt132** [huochaitiantang/pytorch-deep-image-matting#18]
Thanks for the great work! I tried to resume training from the pre-trained model (epoch 12, stage 0) by incorporating the latest changes you made. However, the loss does not seem to improve. Would you share with us the hyperparameters you used during training? Do you plan to release the new pre-trained models which incorporated the latest changes (erosion, etc.)? Thanks!

**Comment by yxt132** [comment]
also, should I re-size the images to to be consistent with the training setting (e.g. 320x320) or at least close to it to get good results?

**Comment by huochaitiantang** [comment]
I just trained the new model from the beginning rather than using the pre-trained model(epoch 12, stage0). While testing you should choose the "whole" method.

**Comment by yxt132** [comment]
Thanks for your response! Did you use batch size =1 and learning rate = 1e-5?

**Comment by yxt132** [comment]
I was able to get similar results as yours for stage 0 now. will try later stages now. I am also thinking to add gradient loss into the loss function to see if better results can be achieved. Please keep us posted on your progress on the stage 1 training. We can compare notes.

**Comment by FantasyJXF** [comment]
@yxt132 HI, I'd like to ask about the test processdure. The Abode datasets fg imgs all have a shadow around the object, so what is for? and how could I test the demo with original image and trimap?

could you please share one of your test set?

**Comment by FantasyJXF** [comment]
* original image

![woman](https://user-images.githubusercontent.com/16235526/61114932-f3f4de80-a4c3-11e9-9243-e9abb48752c1.jpg)

* Instance segment people. (I think that's not OK )
![woman_1](https://user-images.githubusercontent.com/16235526/61115121-5817a280-a4c4-11e9-91bb-a62cc6f4dd6a.png)
and get mask
![woman_mask](https://user-images.githubusercontent.com/16235526/61115157-6796eb80-a4c4-11e9-8e31-8e3142d8cc79.png)

* add trimap
![woman](https://user-images.githubusercontent.com/16235526/61115172-77aecb00-a4c4-11e9-96e2-e40d4845bdde.png)

* finally tun the author's model, get alpha image
![woman](https://user-images.githubusercontent.com/16235526/61115200-8ac19b00-a4c4-11e9-88d6-bb3d5c0f1453.png)

* add alpha channel to origin image
![res](https://user-images.githubusercontent.com/16235526/61115467-1a674980-a4c5-11e9-9db2-c2b8fcf12131.png)


Any one who can teach me how to test my own image?

**Comment by yxt132** [comment]
Here is one of the test sets from Adobe:

foreground:
![girl-1219339_1920_0 3](https://user-images.githubusercontent.com/43161328/61193638-5251d500-a6ef-11e9-9d44-c83e039fc217.png)

background:
![girl-1219339_1920_0 2](https://user-images.githubusercontent.com/43161328/61193646-5b42a680-a6ef-11e9-983d-e2c641b43d05.png)

composited image:
![girl-1219339_1920_0 4](https://user-images.githubusercontent.com/43161328/61193653-65fd3b80-a6ef-11e9-857c-15986706550d.png)

alpha:
![girl-1219339_1920_0](https://user-images.githubusercontent.com/43161328/61193659-6bf31c80-a6ef-11e9-989f-5656945eb6b9.png)

Trimap:
![girl-1219339_1920_0 5](https://user-images.githubusercontent.com/43161328/61193662-71e8fd80-a6ef-11e9-9bb9-bfd07cd36a00.png)

**Comment by FantasyJXF** [comment]
@yxt132  the Adobe foreground image are not the original photo. the woman's hair have some shaow, which might be added manually.

How to test the model with any your own images?

**Comment by wrrJasmine** [comment]
@yxt132  did you set the batch size=1 and lr=1e-5 to get the similar result as the author? 
I do not understand the reason to set the batch-size=1. Is it mentioned in the paper?

**Comment by FantasyJXF** [comment]
The paper just mentioned the Learning rate 1e-5.  I used batch size 1 to train, it's very slow, about 34 hours. 

Did you know why the fg images of Adobe dataset have shadows around the object?



发自我的 iPhone

**Comment by wrrJasmine** [comment]
@yxt132 不好意思，我也不知道为什么前景图片会有这样的阴影
我看你之前提到你无法复现作者开源的精度，后面又能跑出与作者差不多的精度，这期间是做了什么操作吗，我现在也跑不出作者开源的差不多的精度，还有，还想再问一下batch size要设为1的理由

**Comment by yxt132** [comment]
The shades don't matter as long as you have good alpha channel. I did not do much but just used the author's latest code and get similar performance. I guess the batch size is 1 mainly because of the gpu memory limitation. The author is in better position to answer this question.

**Comment by FantasyJXF** [comment]
> The shades don't matter as long as you have good alpha channel. I did not do much but just used the author's latest code and get similar performance. I guess the batch size is 1 mainly because of the gpu memory limitation. The author is in better position to answer this question.

But if the original fg has shades, it will influence the total loss
